openlibrary/core/cache.py

```diff
@@ -160,7 +160,9 @@ def join_threads(self):
         for name, thread in self.active_threads.items():
             thread.join()
 
-    def encode_args(self, args, kw={}):
+    def encode_args(self, args, kw=None):
+        if kw is None:
+            kw = kw or {}
```

Comment:
```suggestion
        kw = kw or {}
```
Repeat with all other instances.